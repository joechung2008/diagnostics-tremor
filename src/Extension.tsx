import Configuration from "@/Configuration";
import StageDefinition from "@/StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="flex flex-col gap-2 max-h-max overflow-y-auto p-1 grow">
      <h2 className="font-semibold text-3xl">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
