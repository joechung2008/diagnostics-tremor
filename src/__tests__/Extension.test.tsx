import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Extension from "../Extension";

describe("Extension", () => {
  it("should render correctly with all props", () => {
    const { container } = render(
      <Extension
        extensionName="Test Extension"
        config={{ key1: "value1", key2: "value2" }}
        stageDefinition={{
          stage1: ["step1", "step2"],
          stage2: ["step3"],
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render correctly without config", () => {
    const { container } = render(
      <Extension
        extensionName="Test Extension"
        stageDefinition={{
          stage1: ["step1", "step2"],
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render correctly without stageDefinition", () => {
    const { container } = render(
      <Extension extensionName="Test Extension" config={{ key1: "value1" }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render correctly with minimal props", () => {
    const { container } = render(<Extension extensionName="Test Extension" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
