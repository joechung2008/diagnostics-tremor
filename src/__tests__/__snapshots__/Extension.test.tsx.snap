// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > should render correctly with all props 1`] = `
<div
  class="flex flex-col gap-2 max-h-max overflow-y-auto p-1 grow"
>
  <h2
    class="font-semibold text-3xl"
  >
    Test Extension
  </h2>
  <div>
    <h3
      class="text-2xl"
    >
      Configuration
    </h3>
    <div>
      <div
        class="w-full overflow-auto whitespace-nowrap"
      >
        <table
          aria-label="Configuration"
          class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
          tremor-id="tremor-raw"
        >
          <thead
            class=""
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Key
              </th>
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="divide-y divide-gray-200 dark:divide-gray-800"
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                key1
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                value1
              </td>
            </tr>
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                key2
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                value2
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div>
    <h3
      class="text-2xl"
    >
      Stage Definitions
    </h3>
    <div>
      <div
        class="w-full overflow-auto whitespace-nowrap"
      >
        <table
          aria-label="Stage Definitions"
          class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
          tremor-id="tremor-raw"
        >
          <thead
            class=""
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Key
              </th>
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="divide-y divide-gray-200 dark:divide-gray-800"
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                stage1
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                step1, step2
              </td>
            </tr>
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                stage2
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                step3
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Extension > should render correctly with minimal props 1`] = `
<div
  class="flex flex-col gap-2 max-h-max overflow-y-auto p-1 grow"
>
  <h2
    class="font-semibold text-3xl"
  >
    Test Extension
  </h2>
</div>
`;

exports[`Extension > should render correctly without config 1`] = `
<div
  class="flex flex-col gap-2 max-h-max overflow-y-auto p-1 grow"
>
  <h2
    class="font-semibold text-3xl"
  >
    Test Extension
  </h2>
  <div>
    <h3
      class="text-2xl"
    >
      Stage Definitions
    </h3>
    <div>
      <div
        class="w-full overflow-auto whitespace-nowrap"
      >
        <table
          aria-label="Stage Definitions"
          class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
          tremor-id="tremor-raw"
        >
          <thead
            class=""
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Key
              </th>
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="divide-y divide-gray-200 dark:divide-gray-800"
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                stage1
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                step1, step2
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Extension > should render correctly without stageDefinition 1`] = `
<div
  class="flex flex-col gap-2 max-h-max overflow-y-auto p-1 grow"
>
  <h2
    class="font-semibold text-3xl"
  >
    Test Extension
  </h2>
  <div>
    <h3
      class="text-2xl"
    >
      Configuration
    </h3>
    <div>
      <div
        class="w-full overflow-auto whitespace-nowrap"
      >
        <table
          aria-label="Configuration"
          class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
          tremor-id="tremor-raw"
        >
          <thead
            class=""
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Key
              </th>
              <th
                class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="divide-y divide-gray-200 dark:divide-gray-800"
          >
            <tr
              class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
            >
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                key1
              </td>
              <td
                class="p-4 text-sm text-gray-600 dark:text-gray-400"
              >
                value1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
